pkg/controllers/networking/pod_controller.go

```diff
@@ -171,53 +171,60 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result
 		return ctrl.Result{}, wrapError("unable to decouple pod", r.decouple(ctx, pod))
 	}
 
-	var handledByWebhook = globalutils.ParseBoolOrDefault(pod.Annotations[constants.AnnotationHandledByWebhook], false)
+	var (
+		networkStrFromWebhook  string
+		subnetStrFromWebhook   string
+		networkTypeFromWebhook types.NetworkType
+		ipFamilyFromWebhook    types.IPFamilyMode
+		ipFamily               types.IPFamilyMode
+	)
 
-	ipFamily := ipamtypes.ParseIPFamilyFromString(pod.Annotations[constants.AnnotationIPFamily])
+	var handledByWebhook = globalutils.ParseBoolOrDefault(pod.Annotations[constants.AnnotationHandledByWebhook], false)
+	// parse network and network-type in the webhook way
 	if !handledByWebhook {
-		switch ipFamily {
-		case ipamtypes.IPv4, ipamtypes.IPv6, ipamtypes.DualStack:
-		default:
+		if networkStrFromWebhook, _, networkTypeFromWebhook,
+			ipFamilyFromWebhook, _, err = utils.ParseNetworkConfigOfPodByPriority(ctx, r, pod); err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to parse network config of pod: %v", err)
+		}
+
+		ipFamily = ipFamilyFromWebhook
+		if !ipamtypes.IsValidFamilyMode(ipFamily) {
```

Comment:
ip family has been validated in `ParseNetworkConfigOfPodByPriority`, so this line is unreachable

Reply:
done